src/snowflake/connector/query_context_cache.py

```diff
@@ -115,23 +117,26 @@ def _replace_qce(
     ) -> None:
         self._remove_qce(old_qce)
         self._add_qce(new_qce)
+        
+    def sync_priority_map(self):
+        """
+        Sync the _intermediate_priority_map with the _priority_map at the end of the current round of merges.
+        """
+        logger.debug(f"syncPriorityMap called priority_map size = {len(self._priority_map)}, new_priority_map size = {len(self._intermediate_priority_map)}")
+        
+        for key, value in self._intermediate_priority_map.items():
+            self._priority_map[key] = value
+
+        # Clear the _intermediate_priority_map for the next round of QCC merge (a round consists of multiple entries)
+        self._intermediate_priority_map.clear()
 
     def merge(
         self, id: int, read_timestamp: int, priority: int, context: str
     ) -> None:
         if id in self._id_map:
             qce = self._id_map[id]
-            # when id if found in cache and we are operating on a more recent timestamp
-            if read_timestamp > qce.read_timestamp:
-                if qce.priority == priority:
-                    # same priority updates the current context object
-                    qce.read_timestamp = read_timestamp
-                    qce.context = context
```

Comment:
We essentially removed this logic. Is this expected? Just want to make sure this is expected

Reply:
Like I said in the description, because now id_map, priority_map and tree_set saves a `copy.deepcopy` of qce, we can not simply do in-place update like the old logic. We use `replace_qce` instead